File: scratch/update_leads_mock.py
import re
import random

with open('src/mocks/leads.mock.ts', 'r') as f:
    content = f.read()

names = [
  'Abebe Kebede', 'Sara Bekele', 'John Doe', 'Jane Smith', 
  'Mohammed Ali', 'Fatima Nur', 'David Chen', 'Maria Garcia',
  'Tadesse Mengistu', 'Aster Awoke', 'Hassan Ibrahim', 'Amira Said'
]

counter = 0

def replacer(match):
    global counter
    id_val = match.group(1)
    rest = match.group(2)
    
    if 'farmerName:' in rest:
        return match.group(0)
        
    name = names[counter % len(names)]
    farmer_id = f"FID-{random.randint(1000000, 9999999)}"
    consent_date = f"May {random.randint(10, 28)}, 2026"
    counter += 1
    
    return f"{{ id: '{id_val}', farmerName: '{name}', farmerId: '{farmer_id}', consentDate: '{consent_date}',{rest}}}"

new_content = re.sub(r"\{\s*id:\s*'([^']+)',([^}]+)\}", replacer, content)

with open('src/mocks/leads.mock.ts', 'w') as f:
    f.write(new_content)

print("Mock data updated.")

File: scratch/update_leads_mock.js
const fs = require('fs');
const path = require('path');

const filePath = path.join(__dirname, '../src/mocks/leads.mock.ts');
let content = fs.readFileSync(filePath, 'utf8');

// List of realistic names
const names = [
  'Abebe Kebede', 'Sara Bekele', 'John Doe', 'Jane Smith', 
  'Mohammed Ali', 'Fatima Nur', 'David Chen', 'Maria Garcia',
  'Tadesse Mengistu', 'Aster Awoke', 'Hassan Ibrahim', 'Amira Said'
];

let counter = 0;
// We need to match the object literals in the array.
// A regex to match each object in the leadRows array.
const newContent = content.replace(/\{ id: '([^']+)',([^}]+)\}/g, (match, id, rest) => {
  const name = names[counter % names.length];
  const farmerId = 'FID-' + Math.floor(1000000 + Math.random() * 9000000);
  const consentDate = 'May ' + (10 + Math.floor(Math.random() * 20)) + ', 2026';
  counter++;
  
  // If farmerName is already there, don't add it again
  if (rest.includes('farmerName:')) return match;

  return `{ id: '${id}', farmerName: '${name}', farmerId: '${farmerId}', consentDate: '${consentDate}',${rest}}`;
});

fs.writeFileSync(filePath, newContent);
console.log('Mock data updated successfully.');
